test/scripts/add_newline.py

```diff
@@ -0,0 +1,59 @@
+import os, re
+
+
+FILE_PATTERNS = [
+    ".+\.js",
+    ".+\.jsx",
+    ".+\.ts",
+    ".+\.tsx",
+    ".+\.sh",
+    ".+\.json",
+    ".+\.py",
+    ".+\.yml",
+    ".+\.env",
+    ".+\.lua",
+    ".+\.sql",
+    ".+\.sx",
+    ".+\.php",
+    ".+\.xml",
+    ".+\.md",
+    ".+\.template",
+    "Dockerfile",
+]
+EXCLUDE_PATHS = ["node_modules", ".git", "git_cache", "cache", "artifacts"]
```

Comment:
maybe we should add a note that this only works for directories here? or maybe just rename the var to `EXCLUDE_DIRS`?

Reply:
Renamed to EXCLUDE_DIRS